File: frontend/assets/js/ui/health-slot-labels.js
/** 健康聊天槽位的用户文案，内部字段只在接口和状态机中使用。 */
export const HEALTH_SLOT_LABELS = Object.freeze({
    domain: "推荐类型",
    mealTime: "用餐时间",
    mood: "今天的心情",
    scene: "用餐场景",
    healthGoal: "健康目标",
    cuisine: "菜系或食材",
    taste: "口味",
    convenience: "能接受的耗时和购买方式",
    bodyPart: "训练部位",
    bodyParts: "训练部位",
    equipment: "器材",
    goal: "训练目标",
    trainingGoal: "训练目标",
    difficulty: "难度",
    weekStart: "目标周",
    trainingDays: "训练日期",
    timeWindow: "训练时段",
    wakeTime: "起床时间",
    bedtime: "入睡时间",
    sleepDuration: "睡眠时长"
});

/** 将接口字段名转换为普通用户能理解的槽位文案。 */
export function slotLabel(slot) {
    const value = String(slot || "").trim();
    if (!value) return "其他偏好";
    if (HEALTH_SLOT_LABELS[value]) return HEALTH_SLOT_LABELS[value];
    if (/^[\u4e00-\u9fff]/.test(value) && !/[A-Za-z]/.test(value)) return value;
    return "其他偏好";
}

/** 将已确认摘要中的内部字段名替换为中文，保留对应的用户值。 */
export function formatSlotSummary(summary) {
    const value = String(summary || "").trim();
    if (!value) return "";
    const separator = value.indexOf("：") >= 0 ? "：" : value.indexOf(":") >= 0 ? ":" : null;
    if (!separator) return slotLabel(value);
    const index = value.indexOf(separator);
    const key = value.slice(0, index).trim();
    const detail = value.slice(index + separator.length).trim();
    return detail ? `${slotLabel(key)}：${detail}` : slotLabel(key);
}

File: src/main/java/com/diet/health/intent/HealthSlotLabels.java
package com.diet.health.intent;

import java.util.Map;

/** 健康槽位的用户可见中文文案，内部字段只用于协议和状态机。 */
public final class HealthSlotLabels {

    private static final Map<String, String> LABELS = Map.ofEntries(
            Map.entry("domain", "推荐类型"),
            Map.entry("mealTime", "用餐时间"),
            Map.entry("mood", "今天的心情"),
            Map.entry("scene", "用餐场景"),
            Map.entry("healthGoal", "健康目标"),
            Map.entry("cuisine", "菜系"),
            Map.entry("foodType", "餐食类型"),
            Map.entry("taste", "口味"),
            Map.entry("convenience", "能接受的耗时和购买方式"),
            Map.entry("bodyPart", "训练部位"),
            Map.entry("bodyParts", "训练部位"),
            Map.entry("equipment", "器械"),
            Map.entry("trainingGoal", "训练目标"),
            Map.entry("difficulty", "难度"),
            Map.entry("trainingDays", "训练日"),
            Map.entry("timeWindow", "训练时段"),
            Map.entry("wakeTime", "起床时间"),
            Map.entry("bedtime", "入睡时间"),
            Map.entry("sleepDuration", "睡眠时长")
    );

    private HealthSlotLabels() {
    }

    /** 未知字段不直接回显，避免内部协议名进入用户文案。 */
    public static String label(String slot) {
        if (slot == null || slot.isBlank()) {
            return "其他偏好";
        }
        return LABELS.getOrDefault(slot, "其他偏好");
    }
}
